src/embedding.jl

```diff
@@ -122,70 +104,88 @@ function embedGigaSOM(som::GigaSOM.Som, data::Array{Any,1};
     end
 
     # prepare the kNN-tree for lookups
-    tree = knnTreeFun(Array{Float64,2}(transpose(som.codes)), metric)
+    tree = knnTreeFun(Matrix{Float64}(transpose(som.codes)), metric)
 
     # run the distributed computation
     nWorkers = nprocs()
     if nWorkers > 1
-
-        dRes = [ (@spawnat pid embedGigaSOM_internal(som, data[p], tree,
-                                                   k, adjust, boost))
+        dData = distribute(Matrix{Float64}(data))
+        dRes = [ (@spawnat pid embedGigaSOM_internal(som, localpart(dData), tree,
+                                                   k, adjust, boost, m))
                  for (p,pid) in enumerate(workers()) ]
 
         return vcat([fetch(r) for r in dRes]...)
     else
-        return embedGigaSOM_internal(som, data, tree, k, adjust, boost)
+        return embedGigaSOM_internal(som, data, tree, k, adjust, boost, m)
     end
 end
 
 """
-    embedGigaSOM_internal(som::GigaSOM.Som, data::Array{Float64,2}, tree,
+    embedGigaSOM_internal(som::GigaSOM.Som, data::Matrix{Float64}, tree,
```

Comment:
What is actually the difference between `Array{Float64,2}` and  `Matrix{Float64}`? Should we have a `Union` of both?

Reply:
AFAIK `Matrix{...}` should be a type alias for the `Array{..., 2}`, for me it seems better to use the high-level names.